File: version-2/php-ofc-library/y_axis.php
<?php

class y_axis
{
	function y_axis()
	{
		$this->stroke      		= 2;
		$this->tick_length      = 5;
		$this->colour      		= "#d09090";
		$this->grid_colour      = "#00ff00";
		//$this->offset           = 1;
		//$this->labels      		= array( "slashdot.org","digg.com","reddit.com" );
	}
	
	function set_range( $min, $max )
	{
		$this->min = $min;
		$this->max = $max;
	}
	
	function set_offset( $off )
	{
		$this->offset = $off?1:0;	
	}
	
	function set_labels( $labels )
	{
		$this->labels = $labels;	
	}
}

File: version-2/dot-net-library/written-by-xiao-yifang/OpenFlashChart/YAxis.cs
using System;
using System.Collections.Generic;
using System.Text;
using JsonFx.Json;


namespace OpenFlashChart
{
    public class YAxis:Axis
    {
        private int offset;
        private int tick_length;
        private IList<String> labels;
        [JsonProperty("tick-length")]
        public int TickLength
        {
            get { return tick_length; }
            set { tick_length = value; }
        }
        public void SetRange(double min, double max)
        {
            base.Max = max;
            base.Min = min;
        }
        public void SetRange(double min, double max, int step)
        {
            base.Max = max;
            base.Min = min;
            base.Steps = step;
        }
        [JsonProperty("offset")]
        public int Offset
        {
            get { return offset; }
            set
            {
                offset = value>0 ? 1 : 0;
            }
        }
        [JsonProperty("labels")]
        public IList<string> Labels
        {
            get { return labels; }
            set { labels = value; }
        }
        public void SetLabels(IList<string> labelsvalue)
        {
            if(labels==null)
                labels = new List<string>();
            foreach (string str in labelsvalue)
            {
                labels.Add(str);
            }
        }
       
    }
}
